### PR TITLE
Add Thorlabs APT controller driver

### PR DESCRIPTION
**Pull request by sylvainpelissier**
According to https://www.thorlabs.com/software/apt/APT_Communications_Protocol_Rev_15.pdf documentation.

### CODE REVIEW

**Comment by natezb** [comment]
Thanks for the pull request, I really do appreciate it! However, is there some reason you're using APT instead of the Thorlabs Kinesis software? I think it can support everything APT did. We do have support for some individual Kinesis hardware on `master`, but there is also a separate `kinesis` branch for development of a unified Kinesis interface. Development on that is stalled at the moment (due to lack of developer bandwidth), although I do use the branch every day to control my K10CR1 rotation stages.

You can check out #47 for more details on this if you're interested.

**Comment by sylvainpelissier** [comment]
Hi, thanks for the answer. I use APT commands to have a support on Linux machine (no dll required) and not to have to change my code between WIndows and Linux machine.

**Comment by natezb** [comment]
I hadn't realized Kinesis was Windows-only, I think that's suitable justification to include this driver. I'll take a deeper look tomorrow when I get a chance.

File: instrumental/drivers/motion/apt.py [hunk 1]
```diff
@@ -0,0 +1,99 @@
+import serial
+import time
+
+from serial import Serial
+from binascii import unhexlify, hexlify
+from serial.tools.list_ports import comports
+from instrumental.drivers.motion import Motion
```
[the remark this answers is not in the document]
Comment (sylvainpelissier) [reply instrumental/drivers/motion/apt.py]: Agree, and it is no even needed. I deleted it in the following commit.

File: instrumental/drivers/motion/apt.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        return rsp
+
+class TDC001_APT(APT):
+    """
+        Controlling Thorlabs TDC001 T-Cube DC Servo Motor Controllers using APT commands
+    """
+
+    _INST_PARAMS_ = ['serial']
+    
+    pos_scaling_factor = Q_(1/34304, u("mm"))
```
[the remark this answers is not in the document]
Comment (sylvainpelissier) [reply instrumental/drivers/motion/apt.py]: Corrected.

File: instrumental/drivers/motion/apt.py [hunk 3]
```diff
@@ -0,0 +1,99 @@
+import serial
+import time
+
+from serial import Serial
+from binascii import unhexlify, hexlify
```
[the remark this answers is not in the document]
Comment (sylvainpelissier) [reply instrumental/drivers/motion/apt.py]: Deleted it in the following commit.

File: instrumental/drivers/motion/apt.py [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import serial
```
[the remark this answers is not in the document]
Comment (sylvainpelissier) [reply instrumental/drivers/motion/apt.py]: Deleted it in the following commit.

**Comment by sylvainpelissier** [comment]
I agree with you but I am not sure to see how to cache the `Serial` objects within the `APT` class.

**Comment by natezb** [comment]
I'll merge this in, make a few changes, and notify you in this thread so you can check them out and test them.

**Comment by natezb** [comment]
Ok, I cleaned up the driver a bit and added caching of the serial devices. Thanks for the contribution!

For future reference (if someone comes along who has a card-slot type system), I think the right way to implement `list_instruments` looks like this:
- Find all comports with the right VID:PID
- Figure out whether the device is a generic USB unit or a "card-slot" type system. I'm not sure how to do this
- If it's a card-slot type system, send it a message to give the addresses and information of all of its cards, create a `ParamSet` for each card
- Otherwise, create a single `ParamSet` for the generic USB unit

**Pull request merged**